sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -251,7 +251,7 @@ namespace Azure { namespace Core { namespace Http {
     uint16_t m_port{0};
     std::string m_encodedPath;
     // query parameters are all encoded
-    std::map<std::string, std::string> m_encodedQueryParameters;
+    CaseInsensitiveMap m_encodedQueryParameters;
```

Comment:
Keys of query parameters are case-sensitive according to RFC

Reply:
I am not sure if we'd still want to distinguish between them (other than preserving the case), but at the very least this can be changed later without breaking the compile-time compatibility. I reverted this change.